plugins/modules/win_psmodule_info.py

```diff
@@ -0,0 +1,439 @@
+    project_uri:
+      description:
+        - The address of the module's project.
+      type: str
+      sample: 'https://github.com/psake/psake'
+    repository_source_location:
+      description:
+        - The source location of the repository where the module was installed from.
+      type: str
+      sample: 'https://www.powershellgallery.com/api/v2'
+    repository:
+      description:
+        - The PSRepository where the module was installed from.
+        - This value is not historical. It depends on the PSRepositories that are registered now for the current user.
+        - The C(repository_source_location) must match the current source location of a registered repository to get a repository name.
+        - If there is no match, then this value will match C(repository_source_location).
+      type: str
+      sample: PSGallery
+    release_notes:
+      description:
+        - The module's release notes. This is a free text field and no specific format should be assumed.
+      type: str
+      sample: |
+        ## 1.4.6
+        - Update `HelpInfoUri` to point to the latest content
+
+        ## 1.4.5
+        - Bug fix for deadlock when getting parameters in an event
+
+        ## 1.4.4
+        - Bug fix when installing modules from private feeds
+    installed_date:
+      description:
+        - The date the module was installed.
+      type: str
+      sample: '2018-02-14T17:55:34.9620740-05:00'
+    published_date:
+      description:
+        - The date the module was published.
+      type: str
+      sample: '2017-03-15T04:18:09.0000000'
+    updated_date:
+      description:
+        - The date the module was last updated.
+      type: str
+      sample: '2019-12-31T09:20:02.0000000'
+    log_pipeline_execution_details:
+      description:
+        - Determines whether pipeline execution detail events should be logged.
+      type: bool
+    module_list:
+      description:
+        - A list of modules packaged with this module.
+        - This value is not often returned and the modules are not automatically processed.
+      type: list
+      elements: dict
+      contains:
+        name:
+          description:
+            - The name of the module.
+            - This may also be a path to the module file.
+          type: str
+          sample: '.\WindowsUpdateLog.psm1'
+        guid:
+          description:
+            - The GUID of the module.
+          type: str
+          sample: 82fdb72c-ecc5-4dfd-b9d5-83cf6eb9067f
+        version:
+          description:
+            - The minimum version of the module.
+          type: str
+          sample: '2.0'
+        maximum_version:
+          description:
+            - The maximum version of the module.
+          type: str
+          sample: '2.9'
+        required_version:
+          description:
+            - The exact version of the module required.
+          type: str
+          sample: '3.1.4'
+    nested_modules:
```

Comment:
A quick check on this it seems like `nested_modules` is just a dict in the format

```
nested_modules:
  Count: x
  Value:
  - name: abc
    ...
  - name: def
    ...
```

Seems like we need to specially handle whatever this type is in .NET so it's actually serialized as a list.

Reply:
Did you test this from within ansible or by running it directly in PowerShell?

I got those values outside of Ansible, but not within it, strangely enough. I had a workaround prepared and everything and ended up removing it

It would probably wouldn't hurt anything to put it back; basically old redundant type data that's added has to be removed:
https://stackoverflow.com/a/48858780/3905079

Reply:
It was run within Ansible.

Reply:
Ok I'm going to put that workaround back in then since it doesn't seem to hurt anything.

Weirdly it didn't serialize like that in shippable either (the tests would have failed), I wonder what difference it would make being run from Ansible sometimes vs others.

Reply:
@jborean93 pushed the change, let's see if that fixes the serialization for you

Reply:
Looks good, coming up as a list of dicts for me now.